### ISSUE TITLE
Add vehicle leasing component

### ISSUE THREAD

**Issue by mkmahajan** [EnergyInnovation/eps-us#350]
We currently track new vehicle sales and total stock. We should consider adding a leasing component so we can report on vehicle registrations rather than only sales. This may look like a nested logit structure to capture the decision making around sale/lease of vehicles, then vehicle technology.

**Comment by robbieorvis** [comment]
FYI here is an Excel based example of how to implemented hierarchical logits. 
[XCHOICE_Process_Calculation.xlsx](https://github.com/user-attachments/files/18400356/XCHOICE_Process_Calculation.xlsx)

**Comment by robbieorvis** [comment]
I added in some more detail with an example of leasing vs buying for ICE and EVs
[XCHOICE_Process_Calculation.xlsx](https://github.com/user-attachments/files/18400938/XCHOICE_Process_Calculation.xlsx)

**Comment by dobrien13** [comment]
Data on leased vs purchased new vehicles:
https://www.bts.gov/content/new-and-used-passenger-car-sales-and-leases-thousands-vehicles
